libbindgen/src/codegen/mod.rs

```diff
@@ -723,6 +723,17 @@ impl CodeGenerator for CompInfo {
             return;
         }
 
+        // generate an opaque enum if struct is a forward declaration
+        if self.kind() == CompKind::Struct && self.is_forward_declaration() {
```

Comment:
Given that, let's skip also structs with template parameters.

Why only check for `Struct`? instead of both struct and union?